src/bodypart.h

```diff
@@ -107,6 +107,39 @@ struct stat_hp_mods {
 
 struct body_part_type {
     public:
+        /**
+         * the different types of body parts there are.
+         * this allows for the ability to group limbs or determine a limb of a certain type
+         */
+        enum class type {
+            // vital organs
+            vital,
+            // this is where helmets go, and is a vital part.
+            head,
+            // the torso is generally the center of mass of a creature
+            torso,
+            // provides sight
+            sensor,
+            // you eat and scream with this
+            mouth,
+            // may manipulate objects to some degree, is a main part
+            arm,
+            // manipulates objects. usually is not a main part.
+            hand,
+            // provides motive power
+            leg,
+            // helps with balance. usually is not a main part
+            foot,
+            // may reduce fall damage
+            wing,
+            // may provide balance or manipulation
+            tail,
+            // more of a general purpose limb, such as horns.
+            extremity,
```

Comment:
Maybe call it `others`? Because extremity make it seems like we're modeling fingers and toes.